### ISSUE TITLE
VTK writer crashes in parallel when creating vtm file for XML output.

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#1877]
avtVTKWriter::WriteRootFile fails crashes parallel because the fileNames vector used only represents the local processor's filenames rather than the fileNames generated on all ranks. The nblocks value is the total of chunk files across all ranks so the code walks off the end of the fileNames vector. The code needs to gather all fileNames in parallel since it looks like the types of files could possibly vary.

1. Run on 2 processors
2. Open multi_rect3d.silo
3. Add a Pseudocolor plot of u and a Slice operator.
4. Draw plots
5. Export to VTK, turn on the "XML" option.

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Brad Whitlock
Original creation: 10/06/2015 04:51 pm
Original update: 10/07/2015 11:45 am
Ticket number: 2409
